runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/util/SchemaCoderCloudObjectTranslator.java

```diff
@@ -61,11 +62,15 @@ public CloudObject toCloudObject(SchemaCoder target, SdkComponents sdkComponents
         FROM_ROW_FUNCTION,
         StringUtils.byteArrayToJsonString(
             SerializableUtils.serializeToByteArray(target.getFromRowFunction())));
-    Structs.addString(
-        base,
-        SCHEMA,
-        StringUtils.byteArrayToJsonString(
-            SchemaTranslation.schemaToProto(target.getSchema(), true).toByteArray()));
+
+    try {
+      Structs.addString(
+          base,
+          SCHEMA,
+          JsonFormat.printer().print(SchemaTranslation.schemaToProto(target.getSchema(), true)));
+    } catch (Exception e) {
+      throw new RuntimeException(e);
+    }
```

Comment:
`CloudObjectsTest` should be exercising this right?

Reply:
yes